### PR TITLE
Shell script and gradle-profiler scenarios for memory restricted benchmarking

### PR DESCRIPTION
**Pull request by akerimsenol**

### CODE REVIEW

File: apply_diff.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# usage `./apply_diff.sh 500`
+MODULE_COUNT=$1
+declare -A MAX_MEMORY=(
+  ["50"]="400m"
+  ["100"]="600m"
+  ["200"]="1300m"
+  ["500"]="4000m"
+  ["1000"]="9000m"
```
Comment (big-guy) [inline apply_diff.sh]: We've measured these as working with the latest 8.0 nightly:

200 - 1100m
500 - 3200m
1000 - 6800m

File: apply_diff.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# usage `./apply_diff.sh 500`
+MODULE_COUNT=$1
+declare -A MAX_MEMORY=(
+  ["50"]="400m"
+  ["100"]="600m"
```
Comment (big-guy) [inline apply_diff.sh]: Maybe for these two, we should do something slightly different.  400m is below the default memory settings for Gradle (512m). I think it's less interesting to try to fit into a smaller footprint than it is for testing that the larger builds fit into a "reasonable" footprint. 

Could we try these two as "default arguments" and remove the jvmargs from gradle.properties entirely?

File: apply_diff.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  ["200"]="1300m"
+  ["500"]="4000m"
+  ["1000"]="9000m"
+  ["app"]="20000m"
+  ["all"]="60g"
```
Comment (big-guy) [inline apply_diff.sh]: I haven't tried either of these with the latest nightly, so let's just start here and see

**Review by big-guy — CHANGES_REQUESTED** [review]
@jvandort @tresat any other comments?

File: apply_diff.sh (hunk 1 above)
Comment (akerimsenol) [reply apply_diff.sh]: Changed it.

File: apply_diff.sh (hunk 2 above)
Comment (akerimsenol) [reply apply_diff.sh]: Done.

File: apply_diff.sh (hunk 3 above)
Comment (akerimsenol) [reply apply_diff.sh]: Ack.

**Comment by akerimsenol** [comment]
I've updated the PR accordingly, however this will not be merged for a while since Ivan (owner of the repo) is out of office. Feel free to use the fork directly: https://github.com/akerimsenol/android-extra-large-benchmark/tree/2022.9.bash

**Pull request merged**